saleor/graphql/product/types/products.py

```diff
@@ -1090,37 +1148,54 @@ def resolve_image_by_id(root: ChannelContext[models.Product], _info, *, id):
         return root.node.media.filter(pk=pk).first()
 
     @staticmethod
-    def resolve_media(root: ChannelContext[models.Product], info):
-        return MediaByProductIdLoader(info.context).load(root.node.id)
+    def resolve_media(root: ChannelContext[models.Product], info, sort_by=None):
+        if sort_by is None:
+            sort_by = {
+                "field": MediaChoicesSortField.ID.value,  # type: ignore
+                "direction": "",
+            }
+        return MediaByProductIdLoaderSort(info.context, sort_by).load(root.node.id)
 
     @staticmethod
     def resolve_images(root: ChannelContext[models.Product], info):
         return ImagesByProductIdLoader(info.context).load(root.node.id)
 
     @staticmethod
-    def resolve_variants(root: ChannelContext[models.Product], info):
-        requestor = get_user_or_app_from_context(info.context)
-        has_required_permissions = has_one_of_permissions(
-            requestor, ALL_PRODUCTS_PERMISSIONS
-        )
-        if has_required_permissions and not root.channel_slug:
-            variants = ProductVariantsByProductIdLoader(info.context).load(root.node.id)
-        elif has_required_permissions and root.channel_slug:
-            variants = ProductVariantsByProductIdAndChannel(info.context).load(
-                (root.node.id, root.channel_slug)
-            )
-        else:
-            variants = AvailableProductVariantsByProductIdAndChannel(info.context).load(
-                (root.node.id, root.channel_slug)
+    def resolve_variant(root: ChannelContext[models.Product], info, id=None, sku=None):
```

Comment:
I think that you can make this simpler. Something like this one:
```python
    @staticmethod
    def resolve_variant(root: ChannelContext[models.Product], info, id=None, sku=None):
        if id is None and sku is None:
            return None

        def get_product_variant(variants):
            if id:
                id_type, variant_id = graphene.Node.from_global_id(id,)
                if id_type != "ProductVariant":
                    return None
                variant = [variant for variant in variants if variant.pk == variant_id ]
                return variant[0] if variant else None
            if sku:
                variant = [variant for variant in variants if variant.sku == sku ]
                return variant[0] if variant else None
            return None

        variants = resolve_variants(root, info)
        return variants.then(get_product_variant)
```
No need to pass `keys` to dataloader. The function used in `then` already has an access to `resolve_variant`'s context. 
No need to check both of these fields. Personally I would say that we should extend description that if both fields are provided then we return null.